docs/contributing/Pull_Requests.mdx

```diff
@@ -44,7 +44,21 @@ For rule tests we recommend, when reasonable:
 - Including both `valid` and `invalid` code changes in most PRs that affect rule behavior
 - Limiting to one error per `invalid` case
 
-### Raising the PR
+### AST Testing
+
+AST tests belong in the `ast-spec` package, within `fixtures/` subfolders alongside the AST declarations. Each test has its own `<test-name>/` folder, containing a `fixture.ts` file defining the test, and a `snapshots/` subfolder. Happy-path test folders are stored directly under `fixtures/`; error-path test folders go under `fixtures/_errors/`. You can check out the [`ClassDeclaration` fixtures folder](https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/ast-spec/src/declaration/ClassDeclaration/fixtures) for an example of both happy-path and unhappy-path test folders.
+
+### Updating Snapshots
+
+Jest snapshots are generated for use in some tests, e.g. for [rule schemas](https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/eslint-plugin/tests/schema-snapshots) and [code examples in rule docs](https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/eslint-plugin/tests/docs-eslint-output-snapshots). You may need to re-generate these snapshots after adjusting a rule and/or its documentation, by running the relevant test suite(s) with the `-u` flag:
```

Comment:
> `.../main/...`

[Docs] It's best to use commit permalinks, as file intent and/or paths can change over time. E.g. `https://github.com/typescript-eslint/typescript-eslint/tree/04990d545fc119329551ae3a55d79dfc0c7bf147/packages/eslint-plugin/tests/schema-snapshots`.

Reply:
@JoshuaKGoldberg I slightly prefer linking to `main` in this case:

- If the file is missing, we _want_ to update the docs
- If we updated the code, we don't want anyone to read outdated code.

Reply:
Hmm, I guess that makes sense. I feel more comfortable about these kinds of permalinks when Docusaurus flags on missing links...

Reply:
Happy to do whatever we think is best here!